components/AutoComplete/AutoComplete.jsx

```diff
@@ -163,36 +164,37 @@ const AutoComplete = ({
   );
   const [showSuggestions, setShowSuggestions] = useState(false);
   const [cursor, setCursor] = useState(0);
-
   const wrapperRef = useRef();
   const listOptions = useRef();
-
+  const autoInput = useRef(null);
   const EscapeKeyPressValue = 'Escape';
   const assistiveDescriptionDefault = `Digite uma ou mais letras para expandir os resultados. ${filterSuggestionsLength} estão disponiveis.`;
   const assistiveDescriptionDropDownOpen = `${assistiveDescriptionDefault} ${
     filterSuggestions[cursor]
   } 
   ${cursor + 1} de ${filterSuggestionsLength} está destacado`;
 
-  const filterItens = value =>
+  const filterItems = value =>
     suggestions.filter(suggestion => {
       let option = normalizeChars(suggestion);
       option = normalizeChars(option);
       return option.indexOf(normalizeChars(value.toLowerCase())) > -1;
     });
 
   const handleFilter = value => {
-    const filteredValues = filterItens(value);
+    const filteredValues = filterItems(value);
     setShowSuggestions(!!filteredValues.length);
     setFilterSuggestions(filteredValues);
     setFilterSuggestionsLength(filteredValues.length);
   };
 
-  const handleChange = ({ target }) => {
-    setUserTypedValue(target.value);
-    selectedItem(target.value);
+  const handleChange = value => {
+    setUserTypedValue(value);
+    if (value.length >= 3) {
+      onChange(value);
+    }
```

Comment:
I know that we talked about value length, but I think this isn't necessary, because Implementer could be responsible for control this behavior. On this way we wouldn't create a new prop or fix a value to control when component call onChange. What do you think?